From 6135ace6c60c4ea15d7b78df350b0db879f66e4f Mon Sep 17 00:00:00 2001
From: Potin Lai <potin.lai@quantatw.com>
Date: Tue, 18 Jun 2024 23:44:55 +0800
Subject: [PATCH 101/105] dbus-sensors: setup SatelliteSensor meson build

Setup SatelliteSensor meson build for adding ability to extent source code in meta-openbmc repo.

Change-Id: I3b20e8538743d9a17602c0aa55ff862df30fcd70
---
 meson.build               | 4 ++++
 meson.options             | 6 ++++++
 service_files/meson.build | 1 +
 src/meson.build           | 4 ++++
 4 files changed, 15 insertions(+)

diff --git a/meson.build b/meson.build
index d1f1207..8e49f6d 100644
--- a/meson.build
+++ b/meson.build
@@ -49,5 +49,9 @@ default_deps = [
     uring,
 ]
 
+if get_option('satellite').enabled()
+    add_project_arguments('-DAUTO_GEN_SENSOR_HEADER', language : 'cpp')
+endif
+
 subdir('service_files')
 subdir('src')
diff --git a/meson.options b/meson.options
index 00e6fa9..e173f94 100644
--- a/meson.options
+++ b/meson.options
@@ -101,3 +101,9 @@ option(
     value: 'enabled',
     description: 'Enable SMBPBI sensor.',
 )
+option(
+    'satellite',
+    type: 'feature',
+    value: 'disabled',
+    description: 'Enable Satellite sensor.',
+)
diff --git a/service_files/meson.build b/service_files/meson.build
index 5a9938d..e37d471 100644
--- a/service_files/meson.build
+++ b/service_files/meson.build
@@ -20,6 +20,7 @@ unit_files = [
     ['nvme', 'xyz.openbmc_project.nvmesensor.service'],
     ['psu', 'xyz.openbmc_project.psusensor.service'],
     ['smbpbi', 'xyz.openbmc_project.smbpbisensor.service'],
+    ['satellite', 'xyz.openbmc_project.satellitesensor.service'],
 ]
 
 fs = import('fs')
diff --git a/src/meson.build b/src/meson.build
index 0929ba7..c1c8eb9 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -156,3 +156,7 @@ endif
 if get_option('leakdetector').allowed()
     subdir('leakdetector')
 endif
+
+if get_option('satellite').enabled()
+    subdir('satellite')
+endif
-- 
2.47.1

